lib/repl.js

```diff
@@ -374,6 +376,30 @@ function REPLServer(prompt,
     };
   }
 
+  function filterInternalStackFrames(error, structuredStack) {
+    // search from the bottom of the call stack to
+    // find the first frame with a null function name
+    if (typeof structuredStack !== 'object')
+      return structuredStack;
+    const idx = structuredStack.reverse().findIndex(
+      (frame) => frame.getFunctionName() === null);
+
+    // if found, get rid of it and everything below it
+    structuredStack = structuredStack.splice(idx + 1);
+    return structuredStack;
+  }
+
+  function prepareStackTrace(fn) {
+    return (error, stackFrames) => {
+      const frames = filterInternalStackFrames(error, stackFrames);
+      if (fn) {
+        return fn(error, frames);
+      }
+      frames.push(error);
+      return frames.reverse().join('\n    at ');
```

Comment:
@lance `return frames.reverse()` is sufficient?

Reply:
@princejwesley `Error.prepareStacktrace()` is used to format the stack trace.  It’s not required to return a string, but that is the behavior of the default implementation,  and I think it would be unexpected if users suddenly started getting an array from the `Error.stack` property.